lib/mongo/server/connection_common.rb

```diff
@@ -52,6 +52,28 @@ def connected?
       # @api private
       attr_reader :pid
 
+      # Build a document that should be used for connection handshake.
+      #
+      # @param [Server::AppMetadata] app_metadata Application metadata
+      #
+      # @return [BSON::Document] Document that should be sent to a server
+      #     for handshake purposes.
+      #
+      # @api private
+      def handshake_document(app_metadata)
+        document = if app_metadata.server_api && app_metadata.server_api[:version]
+                     HELLO_DOC
+                   else
+                     LEGACY_HELLO_DOC
+                   end
+        document.merge(app_metadata.validated_document)
+      end
+
+      HELLO_DOC = BSON::Document.new({ hello: 1 }).freeze
```

Comment:
These constants either need to be moved under the `private` declaration, marked `@api private` or documented with docstrings if they are public (which I don't think they should be ).

Reply:
Very good point, moved under `private`.
I suppose there is no need to add `@api private` for constants/methods that are in `private` section. Please let me know if it is still needed.

Reply:
No, if a method has private visibility, it isn't part of the public api and @api annotation is not needed.

Reply:
... which holds for most things in driver. For Mongoid I suppose there could be private methods that are part of public API. BSON would be probably in between since it does muck with stdlib classes.